pygsti/baseobjs/basis.py

```diff
@@ -159,7 +168,7 @@ class Basis(_NicelySerializable):
     # could be static methods.
 
     @classmethod
-    def cast_from_name_and_statespace(cls, name: str, state_space: _StateSpace, sparse=None):
+    def cast_from_name_and_statespace(cls, name: str, state_space: _StateSpace, sparse: Optional[bool] = None) -> Basis:
```

Comment:
Wait ... I thought signature type annotations within a class couldn't use that class' name.

For now I'll write a review assuming this sort of type annotation is allowed by all Python versions we support. If it isn't, then we'll need to revisit.